packages/ckeditor5-ui/src/dialog/dialogview.ts

```diff
@@ -0,0 +1,635 @@
+	CURRENT_ROOT_NW = 'current-root-nw'
+}
+
+const toPx = toUnit( 'px' );
+
+/**
+ * TODO
+ */
+export default class DialogView extends DraggableViewMixin( View ) implements DraggableView {
+	/**
+	 * TODO
+	 */
+	public readonly parts: ViewCollection;
+
+	/**
+	 * TODO
+	 */
+	public readonly children: ViewCollection;
+
+	/**
+	 * TODO
+	 */
+	public headerView?: FormHeaderView;
+
+	/**
+	 * TODO
+	 */
+	public closeButtonView?: ButtonView;
+
+	/**
+	 * TODO
+	 */
+	public actionsView?: DialogActionsView;
+
+	/**
+	 * TODO
+	 */
+	public static defaultOffset: number = 15;
+
+	/**
+	 * TODO
+	 */
+	public readonly contentView: View;
+
+	/**
+	 * TODO
+	 */
+	public readonly keystrokes: KeystrokeHandler;
+
+	/**
+	 * TODO
+	 */
+	public readonly focusTracker: FocusTracker;
+
+	/**
+	 * TODO
+	 */
+	declare public isVisible: boolean;
+
+	/**
+	 * TODO
+	 */
+	declare public isTransparent: boolean;
+
+	/**
+	 * TODO
+	 */
+	declare public isDraggable: boolean;
+
+	/**
+	 * TODO
+	 */
+	declare public wasMoved: boolean;
+
+	/**
+	 * TODO
+	 */
+	declare public className: string | undefined;
+
+	/**
+	 * TODO
+	 */
+	declare public _top: number;
+
+	/**
+	 * TODO
+	 */
+	declare public _left: number;
+
+	/**
+	 * TODO
+	 */
+	private _getCurrentDomRoot: () => HTMLElement;
+
+	/**
+	 * TODO
+	 */
+	private _getViewportOffset: () => EditorUI[ 'viewportOffset' ];
+
+	/**
+	 * TODO
+	 */
+	declare public position: DialogViewPosition;
+
+	/**
+	 * TODO
+	 */
+	private readonly focusables: ViewCollection;
+
+	/**
+	 * TODO
+	 */
+	private readonly focusCycler: FocusCycler;
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( locale: Locale, getCurrentDomRoot: () => HTMLElement, getViewportOffset: () => EditorUI[ 'viewportOffset' ] ) {
+		super( locale );
+
+		const bind = this.bindTemplate;
+
+		this.set( 'isVisible', false );
+		this.set( 'className', '' );
+		this.set( 'isDraggable', false );
+		this.set( 'isTransparent', false );
+		this.set( 'wasMoved', false );
+		this.set( 'position', DialogViewPosition.SCREEN_CENTER );
+		this.set( '_top', 0 );
+		this.set( '_left', 0 );
+		this._getCurrentDomRoot = getCurrentDomRoot;
+		this._getViewportOffset = getViewportOffset;
+
+		this.decorate( 'moveTo' );
+
+		this.children = this.createCollection();
+		this.children.on<CollectionChangeEvent>( 'change', this._updateFocusCycleableItems.bind( this ) );
+
+		this.parts = this.createCollection();
+		this.contentView = this._createContentView();
+
+		this.keystrokes = new KeystrokeHandler();
+		this.focusTracker = new FocusTracker();
+		this.focusables = new ViewCollection();
+		this.focusCycler = new FocusCycler( {
+			focusables: this.focusables,
+			focusTracker: this.focusTracker,
+			keystrokeHandler: this.keystrokes,
+			actions: {
+				// Navigate form fields backwards using the Shift + Tab keystroke.
+				focusPrevious: 'shift + tab',
+
+				// Navigate form fields forwards using the Tab key.
+				focusNext: 'tab'
+			}
+		} );
+
+		this.setTemplate( {
```

Comment:
One thing that I don't dig is that the dialog is always inside this overlay wrapper that stretches on the whole viewport. I'd imagine that these are two seperate elements that are next to each other:

```
<div class="overlay"></div>
<div class="dialog">
	...
</div>
```

Since the overlay does not have background currently, I guess you can still interact with things that are below it. But the structure gets complicated for anyone that would like to somehow customize this solution. OTOH, I am not sure how you can easily do it anyway, so...

@oleq let's discuss it after you are back.